File: src/main/java/com/dersaun/apigestaocontas/domain/dtos/TotalPessoaMesAnoDTO.java
package com.dersaun.apigestaocontas.domain.dtos;

import com.dersaun.apigestaocontas.domain.Pessoa;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.math.BigDecimal;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class TotalPessoaMesAnoDTO implements Serializable {


    private Long idPessoa;
    private String nome;
    private String sobrenome;
    private BigDecimal total;

    private String corBackgroud;
    private String corBorder;

    public TotalPessoaMesAnoDTO(Pessoa pessoa, BigDecimal total) {
        this.idPessoa = pessoa.getId();
        this.nome = pessoa.getNome();
        this.sobrenome = pessoa.getSobrenome();
        this.total = total;
    }
}

File: src/app/dashboard/dtos/total-pessoa-mes-ano.dto.ts
import { Pessoa } from '@app/pessoa/models/pessoa.entity';

export class TotalPessoaMesAnoDTO {
  idPessoa: number;
  nome: string;
  sobrenome: string;
  total: number;
  corBackground: string;
  corBorder: string;

  constructor(pessoa?: Pessoa, total?: number) {
    this.idPessoa = pessoa?.id || 0;
    this.nome = pessoa?.nome || '';
    this.sobrenome = pessoa?.sobrenome || '';
    this.total = total || 0;
  }
}
